trackintel/analysis/location_identification.py

```diff
@@ -0,0 +1,214 @@
+            )
+        )
+    # then hand it to to the filter function if necessary.
+    if pre_filter:
+        f = pre_filter_locations()
+        sps = sps[f]
+    if recipe == "FREQ":
+        return freq_recipe(sps, "home", "work")
+    else:
+        raise ValueError(f"Recipe {recipe} does not exist.")
+
+
+def pre_filter_locations(
+    sps,
+    agg_level="user",
+    thresh_min_sp=10,
+    thresh_min_loc=10,
+    thresh_sp_at_loc=10,
+    thresh_loc_time=1,
+    thresh_loc_period=pd.Timedelta("5h"),
+):
+    """Filter locations and user out that have not enough data to do a proper analysis.
+
+    Parameters
+    ----------
+    sps : GeoDataFrame (as trackintel staypoints)
+        Staypoints with the column "location_id".
+
+    agg_level: {"user", "dataset"}, default "user"
+        The level of aggregation when filtering locations:
+        - 'user' : locations are filtered per-user.
+        - 'dataset' : locations are filtered over the whole dataset.
+
+    thresh_min_sp : int, default 10
+        Minimum staypoints a user must have to be included.
+
+    thresh_min_loc : int, default 10
+        Minimum locations a user must have to be included.
+
+    thresh_sp_at_loc : int, default 10
+        Minimum number of staypoints at a location must have to be included.
+
+    thresh_loc_time : int, default 1
+        Minimum timespan in hour that a user must spend at location for the location.
+
+    thresh_loc_period : pd.Timedelta (or parseable from `pd.to_timedelta`), default 5h
+        Minimum timespan of first to last visit of a user at a location.
+
+    Returns
+    -------
+    pd.Series
+        Boolean Series containing the filter.
+
+    Examples
+    --------
+    >> sps = sps[ti.analysis.pre_filter_locations(sps)]
+    """
+    assert sps.as_staypoints
+    sps = sps.copy()
+
+    # filtering users
+    user = sps.groupby("user_id").nunique()
+    user_sp = user["started_at"] >= thresh_min_sp  # every staypoint should have a started_at -> count
+    user_loc = user["location_id"] >= thresh_min_loc
+    user_filter_agg = user_sp & user_loc
+    user_filter_agg.rename("user_filter", inplace=True)  # rename for merging
+    user_filter = pd.merge(sps["user_id"], user_filter_agg, left_on="user_id", right_index=True)["user_filter"]
+
+    # filtering locations
+    sps["duration"] = sps["finished_at"] - sps["started_at"]
+    if agg_level == "user":
+        groupby_loc = ["user_id", "location_id"]
+    elif agg_level == "dataset":
+        groupby_loc = ["location_id"]
+    else:
+        raise ValueError(f"Unknown agg_level '{agg_level}' use instead {{'user', 'dataset'}}.")
+    loc = sps.groupby(groupby_loc).agg({"started_at": [min, "count"], "finished_at": max, "duration": sum})
+    loc.columns = loc.columns.droplevel(0)  # remove possible multi-index
+    loc.rename(columns={"min": "started_at", "max": "finished_at", "sum": "duration"}, inplace=True)
+    loc["period"] = loc["finished_at"] - loc["started_at"]
```

Comment:
If I understand correctly `period` considers the time span of visits and `duration` considers only the sum of stay time?
Sometimes this gets a bit confusing. Maybe more comments and description inside the code is needed.

Reply:
Added a comment to clarify that.